test/unit/dialects/abstract/query-generator.test.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+const chai = require('chai'),
+  expect = chai.expect,
+  Op = require('../../../../lib/operators'),
+  QueryGenerator = require('../../../../lib/dialects/abstract/query-generator');
+
+require(__dirname + '/../../support');
+
+function quoteIdentifier(identifier) {
+  return identifier;
+}
+
+function getQueryGenerator(sequelize) {
+  return Object.assign(
+    {},
+    QueryGenerator,
+    {options: sequelize.options, _dialect: sequelize.dialect, sequelize, quoteIdentifier}
+  );
+}
+
+describe('QueryGenerator', () => {
+  describe('whereItemQuery', () => {
+    it('should generate correct query for Symbol operators', function() {
+
+      const QG = getQueryGenerator(this.sequelize);
+      QG.whereItemQuery(Op.or, [{test: {[Op.gt]: 5}}, {test: {[Op.lt]: 3}}, {test: {[Op.in]: [4]}}])
+        .should.be.equal('(test > 5 OR test < 3 OR test IN (4))');
+
+      QG.whereItemQuery(Op.and, [{test: {[Op.between]: [2, 5]}}, {test: {[Op.ne]: 3}}, {test: {[Op.not]: 4}}])
+        .should.be.equal('(test BETWEEN 2 AND 5 AND test != 3 AND test != 4)');
+
+    });
+
+    it('should not parse any strings as aliases  operators', function() {
+      const QG = getQueryGenerator(this.sequelize);
+      expect(() => QG.whereItemQuery('$or', [{test: 5}, {test: 3}]))
+        .to.throw();
```

Comment:
I'm getting an error for one of the internal packages expecting a string and getting an object.
I can specify the error but I'm not sure if it really matters for the test.

Reply:
this method could be failing for any reason, we cant be ok with that it throws something. Error should be deterministic and may be specific when trying to use String alias.

Reply:
I can check for the specific error I'm getting.
It is hard to detect where a string alias are being used cause in this context there aren't any string aliases (otherwise the query would work) and there might even be cases where a valid query will be generated but it will not use the operator (it will be parsed as a regular field)